src/core/thread/mle.cpp

```diff
@@ -1446,24 +1448,40 @@ otError Mle::AppendAddressRegistration(Message &aMessage, AddressRegistrationMod
         VerifyOrExit(counter < OPENTHREAD_CONFIG_MLE_IP_ADDRS_TO_REGISTER, OT_NOOP);
     }
 
-    // For sleepy end device, register external multicast addresses to the parent for indirect transmission
-    if (!IsRxOnWhenIdle())
+    // Append external multicast address.
+    // For sleepy end device, register external multicast addresses to the parent for indirect transmission.
+    // Since Thread 1.2, non-sleepy MED would depend on its Thread 1.2 parent as the proxy to perform
+    // Multicast Listener Report, it registers external multicast addresses of scope larger than realm
+    // local as well.
+    while (Get<ThreadNetif>().GetNextExternalMulticast(iterator, address) == OT_ERROR_NONE)
     {
-        uint8_t      iterator = 0;
-        Ip6::Address address;
-
-        // append external multicast address
-        while (Get<ThreadNetif>().GetNextExternalMulticast(iterator, address) == OT_ERROR_NONE)
+        if (IsRxOnWhenIdle())
         {
-            entry.SetUncompressed();
-            entry.SetIp6Address(address);
-            SuccessOrExit(error = aMessage.Append(&entry, entry.GetLength()));
-            length += entry.GetLength();
-
-            counter++;
-            // only continue to append if there is available entry.
-            VerifyOrExit(counter < OPENTHREAD_CONFIG_MLE_IP_ADDRS_TO_REGISTER, OT_NOOP);
+#if (OPENTHREAD_CONFIG_THREAD_VERSION >= OT_THREAD_VERSION_1_2)
+            // For Thread 1.2 MED, skip MA registration if the parent is Thread 1.1 device.
+            if (GetParent().IsThreadVersion2())
+            {
+                break;
```

Comment:
In Spec 5.24.3.3, it seems to me that a Thread 1.2 MED should not skip MA registration even if the parent is a Thread 1.1 device. 
```
Once an MTD is attached to such a Parent, it performs both these registration actions:
• Registering with MLR.req as described in Section 5.24.2.3, Registration as Multicast Listener (for FTDs)
• Registering with MLE Address Registration TLV as described in Section 5.24.3.1, Registration as Multicast Listener.
```
Note that the Spec says **BOTH** actions are required. 
So, 1.2 MED should not skip MA registration if parent is 1.1?

Reply:
Good catch! 

However I don't see any value for a 1.2 MED registers MA to the Thread 1.1 parent. 
On the contrary, it may cause one extra Child Update Request/Response exchange and  may consume one of the limited address storage capability on the  parent side.

Not sure the background of the text in spec, should we file a ticket to Thread  Group? @jwhui

Reply:
Yes, please file a Thread Group JIRA to discuss this.

Reply:
Do we need to choose for this PR for now?  
I would prefer to just sending `MA reg` to 1.1 parent too because it does what Spec. says.

Reply:
@simonlingoogle 
is there any test case in 1.2 TP that requires that 1.2 MED sends CUR to register MA to 1.1 parent?

Reply:
Verified that in the 5.10.20 test case, a MTD is required to register MAs to its 1.1 parent too, which is in conformance with the current Spec.

Reply:
Thanks for the information @simonlingoogle 

Filed BHC-612 to raise the discussion in Thread Group. Also mentioned possible impact on test 5.10.20.

As we are not switching to the public repo for certification recently, I would prefer to keep the code as is before BHC-612 is resolved.

Thoughts?
@jwhui @simonlingoogle

Reply:
I think following the current Spec. actually lead to less code. So I would suggest follow the Spec.

Reply:
I don't have a strong opinion either way. I am fine to implement the spec for now and change when BHC-612 is resolved.

Reply:
One note is that specification suggests 1.2 MED only should register MA with scope larger than 3 when it attaches to 1.2 parent and rely on its parent to do MLR. 

For my understanding, the intention of such design itself is to reduce the unnecessary CUR. And this PR follows the same design principle.

I would personally think there is chance for the spec to distinguish MED and SED when mentioning `MTD` in the paragraph quoted by @simonlingoogle .
Even if BHC-612 is rejected, it would be easier to reduce the code to follow the spec.

If no strong objection, I personally tend to keep this PR as is.